exporter/datadogexporter/metrics_translator.go

```diff
@@ -63,6 +85,89 @@ func mapDoubleMetrics(name string, slice pdata.DoubleDataPointSlice) []datadog.M
 	return ms
 }
 
+// intCounter keeps the value of an integer
+// monotonic counter at a given point in time
+type intCounter struct {
+	ts    uint64
+	value int64
+}
+
+// mapIntMonotonicMetrics maps monotonic datapoints into Datadog metrics
+func mapIntMonotonicMetrics(name string, prevPts *ttlmap.TTLMap, slice pdata.IntDataPointSlice) []datadog.Metric {
+	ms := make([]datadog.Metric, 0, slice.Len())
+	for i := 0; i < slice.Len(); i++ {
+		p := slice.At(i)
+		ts := uint64(p.Timestamp())
+		tags := getTags(p.LabelsMap())
+		key := metricDimensionsToMapKey(name, tags)
+
+		if c := prevPts.Get(key); c != nil {
+			cnt := c.(intCounter)
+
+			// We calculate the time-normalized delta
+			dx := float64(p.Value() - cnt.value)
+			dt := float64(ts-cnt.ts) / 1e9
+
+			if dt <= 0 {
+				// We were given a point older than the one in memory so we drop it
+				// We keep the existing point in memory since it is the most recent
+				continue
```

Comment:
Can this actually happen or are timeseries always ordered? I could not find anything relevant in the spec.